Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -325,51 +325,66 @@ - (void)configureViews
         OWSAssert(self.loadCellContentBlock);
         OWSAssert(self.unloadCellContentBlock);
 
-        // Flush any pending "text" subviews.
-        [self insertAnyTextViewsIntoStackView:textViews];
-        [textViews removeAllObjects];
-
         bodyMediaView.clipsToBounds = YES;
 
         self.bodyMediaView = bodyMediaView;
         bodyMediaView.userInteractionEnabled = NO;
         if (self.isMediaBeingSent) {
+            // TODO:
             bodyMediaView.layer.opacity = 0.75f;
         }
 
-        if (hasThumbnailForBodyMedia) {
-            // The "body media" view casts a shadow "downward" onto adjacent views,
-            // so we use a "proxy" view to take its place within the v-stack
-            // view and then insert the body media view above its proxy so that
-            // it floats above the other content of the bubble view.
-
-            UIView *bodyProxyView = [UIView new];
-            [self.stackView addArrangedSubview:bodyProxyView];
-
-            [self addSubview:self.mediaShadowView];
-            [self.mediaShadowView autoPinEdge:ALEdgeTop toEdge:ALEdgeTop ofView:bodyProxyView];
-            [self.mediaShadowView autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:bodyProxyView];
-            [self.mediaShadowView autoPinEdge:ALEdgeLeading toEdge:ALEdgeLeading ofView:bodyProxyView];
-            [self.mediaShadowView autoPinEdge:ALEdgeTrailing toEdge:ALEdgeTrailing ofView:bodyProxyView];
-
-            [self.mediaShadowView addSubview:self.mediaClipView];
-            [self.mediaClipView autoPinToSuperviewEdges];
-
-            [self.mediaClipView addSubview:bodyMediaView];
-            [bodyMediaView autoPinToSuperviewEdges];
-
-            [self.bubbleView addPartnerView:self.mediaClipView];
-            [self.bubbleView addPartnerView:self.mediaShadowView];
-
-            // TODO: Constants
-            // TODO: What's the difference between an inner and outer shadow?
-            self.mediaShadowView.fillColor = self.bubbleColor;
-            self.mediaShadowView.layer.shadowColor = [UIColor blackColor].CGColor;
-            self.mediaShadowView.layer.shadowOpacity = 0.12f;
-            self.mediaShadowView.layer.shadowOffset = CGSizeMake(0.f, 0.f);
-            self.mediaShadowView.layer.shadowRadius = 0.5f;
+        if (self.hasFullWidthMediaView) {
+            // Flush any pending "text" subviews.
+            [self insertAnyTextViewsIntoStackView:textViews];
```

Comment:
Right - the reason I didn't do this is because we can have more than one "text stack view."

Consider this example: an incoming group message with image + text.

* text stack { group sender name }.
* image (body media) view.
* text stack { body text view, footer }

Reply:
There's a lot of complexity here.  Once the "message cell" and "message bubble view" settle down a little I intend to add some comments.  But things are still in motion.

Reply:
> Right - the reason I didn't do this is because we can have more than one "text stack view."

Ahh... right. This makes sense then. I guess there are two stack views - there's the stack view that we (maybe) build if there are text views, and then there's the stack view we insert it into.

Maybe `insertAnyTextViewsAsStackView` is more clarifying? Maybe not though...